cloud/cloud_scheduler.cc

```diff
@@ -92,10 +110,18 @@ class LocalCloudScheduler : public CloudScheduler {
     }
     std::lock_guard<std::mutex> lk(job_mutex_);
     long local_id = next_local_id_++;
-    auto job = [this, local_id, callback](void* a) {
+    auto wp = this->weak_from_this();
+    auto job = [local_id, callback, wp = std::move(wp)](void* a) {
```

Comment:
Nice use of the shared_ptr/weak_ptr pattern to ensure we don't call a method on a deleted `this`